presto-main/src/main/java/com/facebook/presto/cost/PlanNodeCostEstimate.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.spi.statistics.Estimate;
+
+import java.util.Objects;
+
+import static com.facebook.presto.spi.statistics.Estimate.unknownValue;
+import static com.facebook.presto.spi.statistics.Estimate.zeroValue;
+import static java.util.Objects.requireNonNull;
+
+public class PlanNodeCostEstimate
+{
+    public static final PlanNodeCostEstimate UNKNOWN_COST = new PlanNodeCostEstimate(unknownValue(), unknownValue(), unknownValue());
+    public static final PlanNodeCostEstimate ZERO_COST = PlanNodeCostEstimate.builder().build();
+
+    private final Estimate networkCost;
+    private final Estimate cpuCost;
+    private final Estimate memoryCost;
+
+    private PlanNodeCostEstimate(Estimate networkCost, Estimate cpuCost, Estimate memoryCost)
```

Comment:
There is a `CostComparator` class injected to `PlanOptimizers`. Any optimizer that wants to compare costs, should use it.